options.go

```diff
@@ -84,6 +109,19 @@ func WithRefreshTimeout(t time.Duration) DialerOption {
 	}
 }
 
+// WithIAMAuthn enables IAM DB Authentication. If no token source has been
+// configured, either using WithTokenSource, WithCredentialsFile, or
+// WithCredentialsJSON, the dialer will use the default token source as defined
+// by https://pkg.go.dev/golang.org/x/oauth2/google#FindDefaultCredentialsWithParams.
```

Comment:
Let's make this slightly less explicit incase we add other future token sources and don't remember to update here. 
```suggestion
// WithIAMAuthn enables automatic IAM Database Authentication. If no token source has been
// configured (such as WithTokenSource, WithCredentialsFile, etc), the dialer will 
// use the default token source as defined
// by https://pkg.go.dev/golang.org/x/oauth2/google#FindDefaultCredentialsWithParams.
```

Reply:
Done.